src/java.base/share/classes/javax/crypto/KEM.java

```diff
@@ -539,10 +540,19 @@ public static KEM getInstance(String algorithm)
         List<Provider.Service> list = GetInstance.getServices(
                 "KEM",
                 Objects.requireNonNull(algorithm, "null algorithm name"));
-        if (list.isEmpty()) {
-            throw new NoSuchAlgorithmException(algorithm + " KEM not available");
+        List<Provider.Service> allowed = new ArrayList<>();
+        for (Provider.Service s : list) {
+            if (!JceSecurity.canUseProvider(s.getProvider())) {
```

Comment:
Is there a test that verifies a provider won't be used if it's not signed?

Should there also be a test that verifies that a provider signed with an unknown key is rejected?